[ENH] add Singularity build and usage

pcamach2 opened the pull request:
Updated README.md files to include instructions for building Singularity images of FastSurfer from a Docker base on a container development machine and running FastSurfer in Singularity containers. This supports containerized deployment on systems in which a direct installation is not feasible and Docker is not preferable or not allowed, such as high performance computing clusters.

LeHenschel committed:
Fix conform.py import error with new recon-surf docker (#137)

* Fix conform.py import error with new recon-surf docker
* Added FastSurfer Env

pcamach2 committed:
Add Singularity Install and Run Instructions

Explain Singularity install process and usage notes to facilitate deployment on HPCs

pcamach2 committed:
Create README.md

Explains setup and usage of FastSurfer in Singularity

pcamach2 committed:
Fix typo

Fix typo on line 168

iabraham reviewed (commented):
@pcamach2, just an additional comment that if you have [Singularity Python](https://pypi.org/project/spython/) then you can create a `sif` file by [directly converting](https://stackoverflow.com/a/66499169/6395452) the Dockerfile to Singularity recipe and building the image, thereby skipping setting up Docker or creating a Docker image. When I tried it out I only had to change the `runscript` at the bottom of the Singularity recipe:

```
cd "/fastsurfer"
exec ./run_fastsurfer.sh --help "$@"
```
to 
```
cd "/fastsurfer"
exec ./run_fastsurfer.sh "$@"
```

pcamach2 commented:
@iabraham Thanks for the info! Feel free to add those directions to the README.md files! 
The only caveat I would add is that some systems require sudo to build from a Singularity def recipe file.

m-reuter commented:
Thanks for the singularity info. We waited with merging, as we plan to make official docker images available on dockerhub which would require us to change the readme text. We also considered to either 
- test and support the creation of singularity sif files from dockerfiles (as stated above) 
- or put supported singularity sif files directly 

I will still merge this for now and will re-visit then.

m-reuter commented:
This should be merged into dev, it currently has conflicts (just use the Dockerfile_reconsurf from dev). Then we can merge it. Thanks.

pcamach2 committed:
Use Dockerfile_reconsurf from dev branch

Changes from Ubuntu 16.04 base image recipe (from stable) to the new 20.04 recipe from the dev branch

pcamach2 commented:
Happy to help! Let me know when you have the official docker images hosted and I can adjust the readme text as needed

m-reuter commented:
Thanks. We will update this after pulling it in to describe how to do this using our official dockerhub images for released versions. We keep the alternative approach that you describe here for developers who want to create their own images based on dev or their own changes.
